Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Menu/EditorMenu.cpp

```diff
@@ -51,14 +51,47 @@ namespace AzToolsFramework
         }
 
         int sortKey = sortKeyIterator->second;
+        bool removed = false;
 
         AZStd::erase_if(
             m_menuItems[sortKey],
-            [actionIdentifier](const MenuItem& item)
+            [&, actionIdentifier](const MenuItem& item)
             {
+                removed = true;
                 return item.m_identifier == actionIdentifier;
             }
         );
+
+        if (removed)
+        {
+            m_subMenuToSortKeyMap.erase(actionIdentifier);
+        }
+    }
+
+    void EditorMenu::RemoveSubMenu(AZStd::string menuIdentifier)
+    {
+        auto sortKeyIterator = m_subMenuToSortKeyMap.find(menuIdentifier);
+        if (sortKeyIterator == m_subMenuToSortKeyMap.end())
+        {
+            return;
+        }
+
+        int sortKey = sortKeyIterator->second;
+        bool removed = false;
+
+        AZStd::erase_if(
+            m_menuItems[sortKey],
+            [&, menuIdentifier](const MenuItem& item)
```

Comment:
Q: is this capturing all used vars by reference but `menuModifier` by value Is this intended? I think either `menuModifier` can be omitted, or the capture group replaced with `[&menuModifier]`.

Reply:
+1 on this Q. Mightn't you be able to capture both variables by reference?

Reply:
Not intended, probably caused by iteration :D Fixed on both instances in that file, and took the chance to fix the ToolBar version as well.